doc/nrf/app_dev/device_guides/nrf54l/cryptography.rst

```diff
@@ -0,0 +1,336 @@
+
+* ``PSA_ERROR_ALREADY_EXIST``: One of the required key slots has already been provisioned.
+* ``PSA_ERROR_NOT_SUPPORTED``: Unsupported key type.
+
+Removing or revoking keys from KMU
+==================================
+
+Keys are deleted or revoked using the ``psa_destroy_key`` function.
+Calling the ``psa_destroy_key`` function on keys that have the persistence ``CRACEN_KEY_PERSISTENCE_REVOKABLE``, will mark the associated KMU slots as revoked, preventing the slots from being reused for new keys.
+
+
+Using KMU keys
+==============
+
+Keys stored in the KMU can be used in standard PSA cryptographic operations for encryption, decryption, signing a hash or a message, and verifying a hash or a message, given that the corresponding ``PSA_KEY_USAGE_*`` flags are set.
+
+Keys with the usage schemes Protected (``CRACEN_KMU_KEY_USAGE_SCHEME_PROTECTED``) and Seed (``CRACEN_KMU_KEY_USAGE_SCHEME_SEED``) can push data from the slots directly to CRACEN registers that are not accessible by the CPU.
+These usage schemes are supported for pushing symmetric keys used for cipher operations, and for seeds used by the CRACEN IKG.
+
+Keys with the usage scheme Raw (``CRACEN_KMU_KEY_USAGE_SCHEME_RAW``), such as symmetric keys, are temporarily pushed to a RAM location by the CRACEN driver, and then loaded by CRACEN into the asymmetric engine before running operations like sign or verify.
+
+Key slots with the usage scheme Encrypted (``CRACEN_KMU_KEY_USAGE_SCHEME_ENCRYPTED``) also have to be decrypted to a temporary push location in RAM before they are used by CRACEN, which is handled by the CRACEN driver.
+
+When the application is built with TF-M, this temporary push location is protected inside the secure processing environment to avoid exposing the key material to the non-secure application.
+If TF-M is not used, the keys are pushed to a reserved RAM area at location 0x20000000-0x20000064 (``kmu_push_area``).
+
+You might encounter the following KMU-specific error codes when using the KMU keys:
+
+* PSA_ERROR_INVALID_HANDLE: Attempting an operation on an empty KMU slot.
+* PSA_ERROR_NOT_PERMITTED: Attempting an operation on a revoked key.
+* PSA_ERROR_HARDWARE_FAILURE: The key slot has invalid data.
+* PSA_ERROR_CORRUPTION_DETECTED: The key slot has invalid data.
+
+Configuration
+*************
+
+The CRACEN peripheral does not require any configuration.
+Its operation is ensured by hardware.
+
+To enable KMU support, set the following Kconfig options:
+
+.. code-block:: console
+
+   CONFIG_NRF_SECURITY=y
+   CONFIG_MBEDTLS_PSA_CRYPTO_C=y
+   CONFIG_PSA_CRYPTO_DRIVER_CRACEN=y
+
+As an example, enabling support for the Encrypted key usage scheme (``CRACEN_KMU_KEY_USAGE_SCHEME_ENCRYPTED``) should look as follows:
+
+.. code-block:: console
+
+   CONFIG_PSA_WANT_KEY_TYPE_AES=y
+   CONFIG_PSA_WANT_AES_KEY_SIZE_256=y
+   CONFIG_PSA_WANT_ALG_ECB_NO_PADDING=y
+   CONFIG_PSA_WANT_ALG_CMAC=y
+   CONFIG_PSA_WANT_ALG_SP800_108_COUNTER_CMAC=y
+   CONFIG_PSA_WANT_ALG_GCM=y
```

Comment:
This is a useful way of providing code, but it's totally ignoring the fact that we have this page that documents how to configure Kconfig: https://docs.nordicsemi.com/bundle/ncs-latest/page/nrf/app_dev/config_and_build/kconfig/index.html 

Moreover, using this code-block method makes it impossible to link to the Kconfig options using `:kconfig:option:`, which is something we recommend in guidelines: https://docs.nordicsemi.com/bundle/ncs-latest/page/nrf/dev_model_and_contributions/documentation/styleguide.html#linking

Reply:
Working on it :)

Reply:
Any better now, @greg-fer? :)